drivers/src/soc_ifc.rs

```diff
@@ -365,6 +365,59 @@ impl SocIfc {
             .notif_internal_intr_r()
             .write(|w| w.notif_cmd_avail_sts(true));
     }
+
+    pub fn uds_program_req(&self) -> bool {
+        self.soc_ifc
+            .regs()
+            .ss_dbg_manuf_service_reg_req()
+            .read()
+            .uds_program_req()
+    }
+
+    pub fn set_uds_programming_flow_state(&mut self, in_progress: bool) {
+        self.soc_ifc
+            .regs_mut()
+            .ss_dbg_manuf_service_reg_rsp()
+            .write(|w| w.uds_program_in_progress(in_progress));
+    }
+
+    pub fn set_uds_programming_flow_status(&mut self, flow_succeeded: bool) {
+        self.soc_ifc
+            .regs_mut()
+            .ss_dbg_manuf_service_reg_rsp()
+            .write(|w| {
+                if flow_succeeded {
+                    w.uds_program_success(true);
+                    w.uds_program_fail(false)
+                } else {
+                    w.uds_program_success(false);
+                    w.uds_program_fail(true)
+                }
+            });
+    }
+
+    pub fn uds_seed_dest_base_addr_low(&self) -> u32 {
+        self.soc_ifc.regs().ss_uds_seed_base_addr_l().read()
+    }
+
+    pub fn active_mode(&self) -> bool {
+        self.soc_ifc
+            .regs()
+            .cptra_hw_config()
+            .read()
+            .active_mode_en()
+    }
+
+    pub fn uds_fuse_row_granularity_64(&self) -> bool {
+        let config_val: u32 = self.soc_ifc.regs().cptra_hw_config().read().into();
+        ((config_val >> 6) & 1) != 0
+    }
+
+    pub fn fuse_controller_base_addr(&self) -> u64 {
+        let low = self.soc_ifc.regs().ss_otp_fc_base_addr_l().read();
+        let high = self.soc_ifc.regs().ss_otp_fc_base_addr_h().read();
+        (high as u64) << 32 | low as u64
```

Comment:
Maybe return AxiAddr instead of u64?

Reply:
Let's maintain the return type as u64 since the caller needs to add an offset to the address. If the return type were AxiAddr, the caller would need to convert it back to u64 before adding the offset.